Stresser/RegistryEventHandler.cpp

```diff
@@ -33,7 +39,7 @@ void RegistryEventHandler::onEventRecord(PEVENT_RECORD record)
 	}
 
 	const std::wstring keyName(keyNameProperty->getUnicodeString());
-	for (const auto& artifact : this->m_artifactsToReport)
+	for (const auto& artifact : this->m_artifacts)
```

Comment:
Unsafe access to the data structure.
Use `AutoCriticalSection autoCriticalSection` before accessing this data structure.
If the `ArtifactManager` will `notify` the `subscribers` he will change this structure and this could lead to dead locks.